### PR TITLE
Enable building native installers for Ubuntu17.04

### PR DESCRIPTION
**Pull request by rakeshsinghranchi**
This enables building Ubuntu 17.04 runtime installers. 
- [x] - Build Ubuntu 17.04 docker image with all the required dependencies
- [x] - Get Package Feed userid set for Ubuntu 17.04
- [x] - Get the UserId key entry in Pipeline build Azure Key Vault 
- [x] - Successfully build packages pointing using private build definition and azure blob storage
- [ ] - Verify Feed upload against official build
- [ ] - Verify Official build once available
- [ ] - Enable Repo readme file to link the official packages for consumption

### CODE REVIEW

**Comment by chcosta** [comment]
Looks reasonable to me

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 1]
```diff
@@ -1043,6 +1206,22 @@
     "CliNuGetApiKey": {
       "value": null,
       "isSecret": true
+    },
+    "DistroRid_Ubuntu1704": {
```
Comment (weshaggard) [inline buildpipeline/Core-Setup-Linux-BT.json]: @eerhardt changed how these are done recently so make sure you are following the new pattern.

Comment (eerhardt) [reply buildpipeline/Core-Setup-Linux-BT.json]: [That change](https://github.com/dotnet/core-setup/pull/3291) hasn't been merged yet.  It probably won't get merged until after this change, so don't worry about it.  I'll resolve any conflicts.

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 2]
```diff
@@ -1,11 +1,13 @@
 {
   "build": [
     {
+      "environment": {},
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: What are these changes?  Are they necessary?

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 3]
```diff
@@ -960,7 +1123,7 @@
       "value": "/p:AzureAccountName=$(PB_AzureAccountName) /p:AzureAccessToken=$(PB_AzureAccessToken) /p:ChecksumAzureAccountName=$(PB_ChecksumAzureAccountName) /p:ChecksumAzureAccessToken=$(PB_ChecksumAzureAccessToken) /p:DebRepoUser=$(PB_DebRepoUser) /p:DebRepoServer=$(PB_DebRepoServer) /p:DebRepoPass=$(DEB_REPO_PASSWORD) $(PB_DebianKeys)"
     },
     "PB_DebianKeys": {
-      "value": "/p:DebianId_ubuntu1404-x64=$(PB_DebianId_ubuntu1404-x64) /p:DebianId_debian8-x64=$(PB_DebianId_debian8-x64) /p:DebianId_debian9-x64=$(PB_DebianId_debian9-x64) /p:DebianId_ubuntu1604-x64=$(PB_DebianId_ubuntu1604-x64) /p:DebianId_ubuntu1610-x64=$(PB_DebianId_ubuntu1610-x64)"
+      "value": "/p:DebianId_ubuntu1404-x64=$(PB_DebianId_ubuntu1404-x64) /p:DebianId_debian8-x64=$(PB_DebianId_debian8-x64) /p:DebianId_debian9-x64=$(PB_DebianId_debian9-x64) /p:DebianId_ubuntu1604-x64=$(PB_DebianId_ubuntu1604-x64) /p:DebianId_ubuntu1610-x64=$(PB_DebianId_ubuntu1610-x64)  /p:DebianId_ubuntu1704-x64=$(PB_DebianId_ubuntu1704-x64)"
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: (nit) two spaces in between parameters here.

`$(PB_DebianId_ubuntu1610-x64)  /p:DebianId_ubuntu1704-x64=$(PB_DebianId_ubuntu1704-x64)`

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 4]
```diff
@@ -1043,6 +1206,22 @@
     "CliNuGetApiKey": {
       "value": null,
       "isSecret": true
+    },
+    "DistroRid_Ubuntu1704": {
+      "value": "ubuntu.17.04-$(PB_TargetArchitecture)"
+    },
+    "DockerCommonRunArgs_Ubuntu1704": {
+      "value": "--name $(PB_DockerContainerName)$(DockerTag_Ubuntu1704) -v \\\"$(PB_SourcesDirectory):$(PB_GitDirectory)\\\" -v $(Build.StagingDirectory)/sharedFrameworkPublish/:/root/sharedFrameworkPublish/ -w=\\\"$(PB_GitDirectory)\\\" $(DockerImageName_Ubuntu1704)"
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: Why the triple slashes?  `\\\`?  We don't use them elsewhere.

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 5]
```diff
@@ -1043,6 +1206,22 @@
...
+    },
+    "DockerImageName_Ubuntu1704": {
+      "value": "$(PB_DockerRepository):$(DockerTag_Ubuntu1704)"
+    },
+    "DockerTag_Ubuntu1704": {
+      "value": "ubuntu-17.04-debpkg-7968025-20172512022553"
+    },
+    "PB_DebianId_ubuntu1704-x64": {
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: I assume this secret value is going to get added where necessary.  Correct?

File: buildpipeline/Core-Setup-Linux-BT.json (hunk 2 above)
Comment (rakeshsinghranchi) [reply buildpipeline/Core-Setup-Linux-BT.json]: These are attributes auto-generated by @chcosta tool to update buildpipeline definition files.

File: buildpipeline/Core-Setup-Linux-BT.json (hunk 5 above)
Comment (rakeshsinghranchi) [reply buildpipeline/Core-Setup-Linux-BT.json]: yes, it's added to Key Vault.

File: buildpipeline/Core-Setup-Linux-BT.json (hunk 4 above)
Comment (rakeshsinghranchi) [reply buildpipeline/Core-Setup-Linux-BT.json]: This is again auto-generated by @chcosta 's tool.

**Pull request merged**